File: src/Tokyo.Service/Models/Core/SharedTimeTableInfo.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace Tokyo.Service.Models.Core
{
    public class SharedTimeTableInfo  // TODO security problem: store id of other store is transmitted => use email or some other key/token
    {
        public int SharedTimeTableInfoId { get; set; }

        [Required]
        public string SharedWithTrackerStoreId { get; set; }
        public TrackerStore SharedWithTrackerStore { get; set; }

        [Required]
        public string OwnerTrackerStoreId { get; set; }
        public TrackerStore OwnerTrackerStore { get; set; }

        public int TimeTableId { get; set; }
        public TimeTable TimeTable { get; set; }
        [Required, StringLength(255)]
        public string Name { get; set; }
        public DateTimeOffset ShareEnabledTime { get; set; } // value generated in db
        [Required]
        public bool? IsReshareAllowed { get; set; }
        [Required]
        public bool? IsEditAllowed { get; set; }
    }
}

File: src/Tokyo.Client/Typewriter/SharedTimeTableInfo.ts


interface TrackerStore { TrackerStoreId: string; }
import {TimeTable} from "./TimeTable"; 
export class SharedTimeTableInfo { 
    SharedTimeTableInfoId: number;
    SharedWithTrackerStoreId: string;
    SharedWithTrackerStore: TrackerStore | undefined;
    OwnerTrackerStoreId: string;
    OwnerTrackerStore: TrackerStore | undefined;
    TimeTableId: number;
    TimeTable: TimeTable | undefined;
    Name: string;
    ShareEnabledTime: Date | string
    IsReshareAllowed: boolean;
    IsEditAllowed: boolean;
}